src/handlers/net/NetStreamHandler.cpp

```diff
@@ -403,6 +416,106 @@ void NetworkMetricsBucket::process_dnstap(bool deep, const dnstap::Dnstap &paylo
     }
 }
 
+void NetworkMetricsBucket::process_sflow(bool deep, const SFSample &payload)
+{
+    std::unique_lock lock(_mutex);
+    for (const auto &sample : payload.elements) {
+
+        if (sample.gotIPV4) {
+            ++_counters.IPv4;
+        } else if (sample.gotIPV6) {
+            ++_counters.IPv6;
+        }
+
+        switch (sample.dcd_ipProtocol) {
+        case IP_PROTOCOL::TCP:
+            ++_counters.TCP;
+            break;
+        case IP_PROTOCOL::UDP:
+            ++_counters.UDP;
+            break;
+        default:
+            ++_counters.OtherL4;
+            break;
+        }
+
+        if (sample.ifCounters.ifDirection == DIRECTION::IN) {
+            ++_counters.total_in;
```

Comment:
i believe we'd need to not just `++` but `+=` the amount of packets that the sample saw - it's almost definitely not just one?

Reply:
ifDirection is a enum